### PR TITLE
Allow passing arguments when entry points are used as functions

### LINKED ISSUE #5320: Please provide a better entry point to pyreverse
### Current problem

Currently, I have to do:
```python
import pylint

# Set name.
# Collect filenames in files...

sys.argv = 'pyreverse -o png -p'.split() + [name] + files
try:
    pylint.run_pyreverse()
except SystemExit:
    pass  # pylint always does this.
```

### Desired solution

I would like to do something like:
```python
import pylint

pylint.pyreverse(output=pylint.PNG, project=name, files=files)
```

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :scroll: Docs          |

## Description

@Pierre-Sassoulas  I'm not sure if this warrants a changelog entry? I haven't added one.

I'll do a follow-up for `epylint`, as it requires some additional changes to its `Run` class.

Closes #5320

### CODE REVIEW

**Comment by DanielNoord** [comment]
Oops, I based on the wrong branch. This should be better!

**Review by Pierre-Sassoulas — APPROVED** [review]
Great ! I think a changelog would be bice, if only to advertise the feature. I'll approve either way.

**Comment by DanielNoord** [comment]
How do we call this?

"When invoking pylint, symilar or pyreverse programatically you can now pass an arguments keyword"? I'm really not sure...

**Comment by Pierre-Sassoulas** [comment]
> "When invoking pylint, symilar or pyreverse programatically you can now pass an arguments keyword"? I'm really not sure...

Maybe : "The API of pylint, symilar and pyreverse now do not rely exclusively on ``sys.argv`` and also permits to pass args programmatically" ?

**Review by Pierre-Sassoulas — APPROVED** [review]
👌 You're doing so much lately, you're carrying 2.13.0 on your back 😄

File: pylint/__init__.py [hunk 1]
```diff
@@ -11,17 +11,22 @@
...
 import sys
+from typing import List, Optional
 
 from pylint.__pkginfo__ import __version__
 
 # pylint: disable=import-outside-toplevel
 
 
-def run_pylint():
+def run_pylint(*, arguments: Optional[List[str]] = None):
```
Comment (cdce8p) [inline pylint/__init__.py:21]: I'm probably a bit late on this one.

I would suggest using `args` or `argv` instead, without the keyword requirement. This is what I usually see in these cases.
```py
def main(argv: Optional[Sequence[str]] = None):
    argv = argv or sys.argv[1:]
    ...
```
Keyword only isn't really necessary here, it's just one argument. Even with more, the first will likely always be `args` / `argv`. More detail (optional) once can always be keyword only later, although even then not strictly necessary.

**Pull request merged**